### PR TITLE
Monkeys are now disarmed like humans

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request

Previously, monkey disarming was the old system of RNG stun/lose item. This PR makes disarming monkeys the same as disarming humans.

2020-08-18T04-31-20

## Why It's Good For The Game

1. Consistency. It is surprising that given disarm's massive rework (to the point where a rename was considered), that monkeys still followed the old system.

2. I've seen a surprising amount of new players fight monkeys as practice. They're good foes since they're not sentient (generally) and fight back. This PR lets those new players train their disarm skills.

3. Do you not want to see monkeys shoved into disposals bins?

## Changelog
:cl:
add: Disarming a monkey is now the same as disarming a human.
add: Monkeys can now be shoved into humans and vice versa.
/:cl:

I didn't die to a monkey, I promise.

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 1]
```diff
@@ -231,6 +231,126 @@
...
+	var/obj/structure/table/target_table
+	var/obj/machinery/disposal/bin/target_disposal_bin
+	var/shove_blocked = FALSE //Used to check if a shove is blocked so that if it is knockdown logic can be applied
+
+	//Thank you based whoneedsspace
+	target_collateral_carbon = locate(/mob/living/carbon) in target_shove_turf.contents
+
+	// If we can't shove the target into the carbon (such as if it's an alien), then just pretend nothing was there
+	if (!target_collateral_carbon?.can_be_shoved_into)
+		target_collateral_carbon = null
```
Comment (Jared-Fogle) [inline code/modules/mob/living/carbon/carbon_defense.dm:257]: This should be the only interesting new code in this proc. This used to be `target_collateral_human`.

**Comment by nightred** [comment]
I see a new proc `/mob/living/carbon/proc/disarm` but i do not see documentation for it.

**Comment by ATH1909** [comment]
Can you also make NPC monkeys actually stop moving when flashed? They continue to just crawl around (and even escape your grip!) unless you cuff them, IIRC.

**Comment by IndieanaJones** [comment]
And on top of all that, could you give player monkeys the ability to use disarm?  It's impossible for them to do so currently, even if AI monkeys are capable of doing so.

**Comment by Jared-Fogle** [comment]
These are both fine ideas, but I think they would fit better as separate PRs.

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 2]
```diff
@@ -231,6 +231,9 @@
 		return affecting.body_zone
 	return dam_zone
 
+/// Attempt to disarm the target mob.
+/// Will shove the target mob back, and drop them if they're in front of something dense
+/// or another carbon.
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/carbon_defense.dm:236]: use the `/**` format

File: code/modules/mob/living/carbon/carbon_defense.dm [hunk 3]
```diff
@@ -231,6 +231,131 @@
...
+			var/target_turf = get_turf(target)
+			for(var/obj/O in target_turf)
+				if(O.flags_1 & ON_BORDER_1 && O.dir == shove_dir && O.density)
+					directional_blocked = TRUE
+					break
+			if(target_turf != target_shove_turf) //Make sure that we don't run the exact same check twice on the same tile
+				for(var/obj/O in target_shove_turf)
+					if(O.flags_1 & ON_BORDER_1 && O.dir == turn(shove_dir, 180) && O.density)
+						directional_blocked = TRUE
+						break
```
Comment (Rohesie) [inline code/modules/mob/living/carbon/carbon_defense.dm:293]:
```suggestion
			for(var/obj/obj_content in target_turf)
				if(obj_content.flags_1 & ON_BORDER_1 && obj_content.dir == shove_dir && obj_content.density)
					directional_blocked = TRUE
					break
			if(target_turf != target_shove_turf) //Make sure that we don't run the exact same check twice on the same tile
				for(var/obj/obj_content in target_shove_turf)
					if(obj_content.flags_1 & ON_BORDER_1 && obj_content.dir == turn(shove_dir, 180) && obj_content.density)
						directional_blocked = TRUE
						break
```

**Pull request merged**